### DISCUSSION TITLE
Explanatory illustrations

### DISCUSSION THREAD

**Discussion by jeromekelleher** [tskit-dev/what-is-an-arg-paper#22] [General]
The current draft has two placeholder figures, with the following captions:

> The (unannotated) ARG is closely related to the pedigree graph, where each ``individual'' node is replace by two ``genome'' nodes, and edges between parents and children are only included if there is transfer of ancestral material. We usually only include nodes in an ARG that affect our sample, so that, for example, individuals that are ancestral to only one sampled genome are not included.}

and 

> The specific path through the ARG for a given genome position (and therefore the local genealogical tree) cannot be determined without annotations to the graph. (A) The classical Griffiths approach annotates recombination nodes with the corresponding breakpoint. (B) We can equivalently annotate the \emph{edges} with the genomic intervals carrying ancestral material. Edge annotations are somewhat more general, and have significant computational advantages.}

I initially started with having one figure to explain all of this, but then decided that it was probably too much and would end up being confusing. So, there's two figures now:

- Fig 1 shows the relationship between the ARG *graph* structure with the pedigree
- Fig 2 show the relationship between the different ARG encodings/annotation formats

# The Pedigree and the ARG

The ARG is embedded within the pedigree, but there is some nuance which means it's misleading to say that the ARG is a *subset* of the pedigree:

- Nodes in the pedigree are individuals, whereas nodes in the ARG (as we specifically define it, and as implicitly understood by everyone else from Griffiths onwards) refer to single genomes. Thus, each node in the pedigree (may) correspond to two nodes in the ARG.
- ARGs are usually defined as consisting of CA or RE nodes, so therefore bits of the pedigree in which no common ancestry or recombination occurs (within ancestral material) will be omitted.
- If we want to be able to determine the local trees along the genome, the "raw" graph structure inherited from the pedigree must be annotated with some information about how ancestry was transmitted through the graph (the discussion of these annotations is not for this figure, though I think).

So, terminology wise, we're following Mathieson and Scally in that we're defining the unqualified term "ARG" as just the graph (nodes and edges) that is embedded within the pedigree.

For this figure I'd imagined showing a simple pedigree and how the ARG is emdedded within it. It's not obvious whether this should be one figure showing everything or two panels, one showing the pedigree and one the ARG. Whatever's most effective I think.

# ARG annotations

This figure shows that the graph stucture (under the mathematical understanding of the term, a set of nodes and edges) is not sufficient to encode the local tree relationships along the genome. We need to add some "annotations" to the graph in order to do this. The classical approach is to add a breakpoint to each RE node, and is shown in panel (A). We can equivalently add annotations to the edges (B). We discuss this in detail in the text, and note the generality and computational advantages we get from working with edge annotated ARGs.

In a way, this figure is the most important in the paper, as this is the central message we want to convey. Perhaps we should also add an illustration of the work that's required to generate trees or something? 

@hyanwong, do you have anything lying around that'll get us started on these two figures? Should we keep it as two figures or join the two described here into one multi-panel figure?

**Comment by hyanwong** [comment]
For fig 1 I have the "milk-bottles"-like image which shows not only the pedigree but also the series of cell divisions within individuals, which is, of course, more fundamental. The nice thing about this is that the RE and CA nodes are necessarily different. The bad thing is that it it one level too deep, since in our definitions we refer to an individual without distinguishing the pattern of coalescences *within* an individual (these are normally compressed into a single node).

Here's the image I'm talking about (I have the original in SVG form):

![Screenshot 2022-01-11 at 13 30 40](https://user-images.githubusercontent.com/4699014/148951648-bbb380da-ccc0-4c94-bb48-b39dc2102125.png)

The image in the recent Nielsen paper (https://www.biorxiv.org/content/10.1101/2021.11.15.468686v1 fig 1, esp fig 1A) also springs to mind.

**Reply by hyanwong** [reply]
I agree that 2 images are needed, FWIW.

**Reply by jeromekelleher** [reply]
The downside of this is that it doesn't "look like" a pedigree or an ARG. We want things to be as familiar as possible.

I'm not sure there's much to be gained from going down the to cell level for our target audience (#20)? I think this level of explanation is more suited to an introduction to the concepts, rather than a technical discussion among experts.

**Reply by jeromekelleher** [reply]
Shall we see if anyone is inspired to make a figure in the spirit of the Nielsen preprint (you probably have enough to get on with?)

**Reply by hyanwong** [reply]
> I'm not sure there's much to be gained from going down the to cell level for our target audience (#20)? I think this level of explanation is more suited to an introduction to the concepts, rather than a technical discussion among experts.

Yep, I think you are right, but I think it is worth an aside somewhere that the ARG definition that we are using is actually a simplification of an underlying cellular process.

**Reply by gregorgorjanc** [reply]
Perhaps stick the ~current fig into supplement and create a simplified version for the main paper and state that the ARG "output" is in fact a simplification of the actual/true ARG "output" shown in the supplement?
